File: Codeforces Round 964 (Div. 4)/F_Expected_Median.cpp
#include <bits/stdc++.h>
using namespace std;

const int N = 2e5 + 5, mod = 1e9 + 7;
int64_t fact[N];
int64_t pw(int64_t a, int64_t b)
{
    int64_t r = 1;
    while (b > 0)
    {
        if (b & 1)
            r = (r * a) % mod;
        b /= 2;
        a = (a * a) % mod;
    }
    return r;
}
int64_t C(int64_t n, int64_t k)
{
    if (n < k)
        return 0LL;
    return (fact[n] * pw((fact[n - k] * fact[k]) % mod, mod - 2)) % mod;
}
int main()
{
    int t;
    cin >> t;
    fact[0] = 1;
    for (int64_t i = 1; i < N; ++i)
        fact[i] = (fact[i - 1] * i) % mod;
    while (t--)
    {
        int n, k;
        cin >> n >> k;
        vector<int> a(n);
        int ones = 0;
        for (int i = 0; i < n; ++i)
        {
            cin >> a[i];
            ones += a[i];
        }
        // at least k/2+1 ones
        int64_t ans = 0;
        for (int cnt_ones = k / 2 + 1; cnt_ones <= min(ones, k); ++cnt_ones)
        {
            ans += C(ones, cnt_ones) * C(n - ones, k - cnt_ones) % mod;
            ans %= mod;
        }
        cout << ans << "\n";
    }
}

File: Codeforces Round 964 (Div. 4)/F_Expected_Median.py
MOD = int(1e9 + 7)
N = int(2e5 + 5)
v = [1] * N


def p(a, b, mod=MOD):
    result = 1
    while b > 0:
        if b & 1:
            result = (result * a) % mod
        b //= 2
        a = (a * a) % mod
    return result


def ncr(n, k, mod=MOD):
    if n < k:
        return 0
    return (v[n] * p((v[n - k] * v[k]) % mod, mod - 2)) % mod


for i in range(1, N):
    v[i] = (v[i - 1] * i) % MOD


t = int(input())
for _ in range(t):
    n, k = map(int, input().split())
    a = list(map(int, input().split()))
    ones = sum(a)

  
    ans = 0
    for cnt_ones in range(k // 2 + 1, min(ones, k) + 1):
        ans += ncr(ones, cnt_ones) * ncr(n - ones, k - cnt_ones) % MOD
        ans %= MOD

    print(ans)
